system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/EndToEndApiWithPostgresqlTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.test.e2e;
+
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import org.eclipse.edc.junit.annotations.PostgresqlDbIntegrationTest;
+import org.eclipse.edc.junit.extensions.EdcRuntimeExtension;
+import org.eclipse.edc.spi.persistence.EdcPersistenceException;
+import org.eclipse.edc.sql.testfixtures.PostgresqlLocalInstance;
+import org.eclipse.edc.test.e2e.participant.EndToEndTransferParticipant;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static jakarta.json.Json.createObjectBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.eclipse.edc.connector.transfer.spi.types.TransferProcessStates.TERMINATED;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.TYPE;
+import static org.eclipse.edc.spi.CoreConstants.EDC_NAMESPACE;
+import static org.eclipse.edc.test.e2e.PostgresConstants.JDBC_URL_PREFIX;
+import static org.eclipse.edc.test.e2e.PostgresConstants.PASSWORD;
+import static org.eclipse.edc.test.e2e.PostgresConstants.USER;
+import static org.eclipse.edc.test.system.utils.PolicyFixtures.noConstraintPolicy;
+
+@PostgresqlDbIntegrationTest
+class EndToEndApiWithPostgresqlTest {
```

Comment:
I'm not sure about this test scope, seems to be too broad, I mean, one we ensured that the store filters out the entities by a numeric rightOperand, we're good to go, because all the rest of the interaction is already tested.

TL;DR: I don't think we need this test

Reply:
we don't have any other test, were we actually run a JSON-LD REST request through the connector, until it hits the database.

Now, we could create a separate runtime for management-api-e2e-tests (presumably `management-api-test-runtime-pg`), but I would do it in another PR. It's probably a good idea anyway, even beyond this particular bug.

Reply:
PG implementation should behave the same as the InMemory (see my other comment), to test the transformation thing I would use the `TransferProcessApiEndToEndTest` with the in memory store, given that the two implementations should behave the same (and this will be ensured by putting the `find_queryByState` in the `TestBase` class.

Reply:
yeah, should be the same. creating a PG variant of the management api tests would still be a cheap (in terms of dev cost) and valuable addition i think. 
Here, the particular test case can be moved 👍